src/main/java/uk/gov/pay/directdebit/mandate/params/MandateSearchParams.java

```diff
@@ -65,14 +85,65 @@
         return Optional.ofNullable(toDate).map(ZonedDateTime::parse);
     }
 
-    public int getPage() {
+    public Integer getPage() {
         return page;
     }
 
-    public int getDisplaySize() {
+    public Integer getDisplaySize() {
         return displaySize;
     }
 
+    @Override
+    public String getGatewayExternalId() {
+        return this.gatewayAccountExternalId;
+    }
+
+    @Override
+    public String buildQueryParamString() {
+        var query = new StringBuilder();
+
+        query.append(createQueryParamFor(PAGE, page.toString()));
+        query.append(appendQueryParam(DISPLAY_SIZE, displaySize.toString()));
+
+        if (serviceReference != null) {
+            query.append(appendQueryParam(REFERENCE, serviceReference));
+        }
+
+        if (mandateState != null) {
+            query.append(appendQueryParam(STATE, mandateState.toString()));
+        }
+
+        if (mandateBankStatementReference != null) {
+            query.append(appendQueryParam(BANK_STATEMENT_REFERENCE, mandateBankStatementReference.toString()));
+        }
+
+        if (name != null) {
+            query.append(appendQueryParam(NAME, name));
+        }
+
+        if (email != null) {
+            query.append(appendQueryParam(EMAIL, email));
+        }
+
+        if (fromDate != null) {
+            query.append(appendQueryParam(FROM_DATE, fromDate));
+        }
+
+        if (toDate != null) {
+            query.append(appendQueryParam(TO_DATE, toDate));
+        }
+
+        return query.toString();
+    }
+
+    private String appendQueryParam(String name, String value) {
+        return "&" + createQueryParamFor(name, value);
+    }
+
+    private String createQueryParamFor(String name, String value) {
+        return name + "=" + value;
```

Comment:
Should we be doing URL-encoding here?

Reply:
We follow the same practice in Connector, build a string of parameters which is then passed to a uribuilder which calls an `encode` method with the query String it is provided.

Reply:
I've created a bug ticket https://payments-platform.atlassian.net/browse/PP-5435 for the broader fix.